src/utils/slpReader.ts

```diff
@@ -684,3 +685,49 @@ export function getGameEnd(slpFile: SlpFileType): GameEndType | null {
 
   return gameEndMessage as GameEndType;
 }
+
+export function extractFinalPostFrameUpdates(slpFile: SlpFileType): PostFrameUpdateType[] {
+  const { ref, rawDataPosition, rawDataLength, messageSizes } = slpFile;
+
+  // The following should exist on all replay versions
+  const postFramePayloadSize = messageSizes[Command.POST_FRAME_UPDATE];
+  const gameEndPayloadSize = messageSizes[Command.GAME_END];
+  const frameBookendPayloadSize = messageSizes[Command.FRAME_BOOKEND];
+
+  // Technically this should not be possible
+  if (isUndefined(postFramePayloadSize)) {
+    return [];
+  }
+
+  const gameEndSize = gameEndPayloadSize ? gameEndPayloadSize + 1 : 0;
+  const postFrameSize = postFramePayloadSize + 1;
+  const frameBookendSize = frameBookendPayloadSize ? frameBookendPayloadSize + 1 : 0;
+
+  let frameNum = null;
+  let postFramePosition = rawDataPosition + rawDataLength - gameEndSize - frameBookendSize - postFrameSize;
+  const postFrameUpdates: PostFrameUpdateType[] = [];
+  do {
+    const buffer = new Uint8Array(postFrameSize);
+    readRef(ref, buffer, 0, buffer.length, postFramePosition);
+    if (buffer[0] !== Command.POST_FRAME_UPDATE) {
+      break;
+    }
+
+    const postFrameMessage = parseMessage(Command.POST_FRAME_UPDATE, buffer) as PostFrameUpdateType | null;
+    if (!postFrameMessage) {
+      break;
+    }
+
+    if (isNull(frameNum)) {
```

Comment:
```suggestion
    if (frameNum === null) {
```
nit: why not just compare with null directly here?

Reply:
I kind of like the readability of the first slightly better but it doesn't matter much to me. I can change it.